docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Agent Clustering enables Grafana Agent Flow to coordinate a fleet of agents to
+work together for workload distribution and high availability. It makes agent
+deployments horizontally scalable by default, and minimizes the operational
+overhead of managing your agent infrastructure to handle demand.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model and
+the lightweight [grafana/ckit][] framework. All running nodes are assumed to
+use the same configuration and have access to the same type of hardware and
+network resources such as PVCs and service discovery APIs.
+
+Clustering is built from the ground-up to be the de-facto answer to "how do I
+scale my Grafana Agent setup" in the agent's future. As an example, in
+comparison to a horizontally-scalable setup using [hashmod sharding][],
+clustering with target auto-distribution scales and provides resiliency at
+_half_ the cost in resources and without the need of changing configuration
+files for resharding.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+1-node cluster.
+
+## Usage
+
+The following set of command-line flags are used to configure clustering when
+starting up Grafana Agent:
+- `--server.http.listen-addr`: the address where the agent’s HTTP server
+  listens to.
+- `--cluster.enabled`: enables cluster awareness.
+- `--cluster.node-name`: defines the name used by the cluster node. If not
+  provided, it defaults to the machine's hostname.
+- `--cluster.advertise-address`: defines the address the agent advertises for
+  its peers to connect to. If not provided, it is inferred automatically.
+- `--cluster.join-addresses`: accepts a comma-separated list of addresses to
+  join the cluster at. These can be IP addresses with an optional port, or a
+DNS record to lookup.
+
+Cluster communication happens over HTTP/2 on the agents’ HTTP listener. The
+agents must be configured to accept connections from one another on the address
+defined by the `--server.http.listen-addr` flag.
```

Comment:
Is using `--server.http.listen-addr` accurate if I am using `--cluster.advertise-address`?

Reply:
Nope, you're right, thanks!